### PR TITLE
makefiles/cflags.inc.mk: Use a template for CFLAGS testing

### PR DESCRIPTION
**Pull request by gebart**
### Contribution description

Using a makefile template reduces the code duplication in cflags.inc.mk

### Issues/PRs references

depends on ~#9357~

### CODE REVIEW

**Comment by kYc0o** [comment]
#9357 was merged. Please rebase.

**Comment by gebart** [comment]
rebased

File: makefiles/cflags.inc.mk [hunk 1]
```diff
@@ -25,28 +25,27 @@ ifeq ($(shell $(CC) -fno-delete-null-pointer-checks -E - 2>/dev/null >/dev/null
...
+  $(eval $(call cflags_test_and_add,-fdiagnostics-color))
 endif
 
 # Fast-out on old style function definitions.
 # They cause unreadable error compiler errors on missing semicolons.
 # Worse yet they hide errors by accepting wildcard argument types.
-ifeq ($(shell $(CC) -Wstrict-prototypes -Werror=strict-prototypes -Wold-style-definition -Werror=old-style-definition -E - 2>/dev/null >/dev/null </dev/null ; echo $$?),0)
-  # duplicated parameters don't hurt
-  CFLAGS += -Wstrict-prototypes -Wold-style-definition
-  CXXUWFLAGS += -Wstrict-prototypes -Wold-style-definition
```
Comment (cladmi) [inline makefiles/cflags.inc.mk]: Just for clarity, I would prefer moving the `CXXUWFLAGS` and removing the useless `WERROR` case to a separate "cleanup" commit.

**Comment by kYc0o** [comment]
> Can someone test on mac/llvm ? @kYc0o ?

Works correctly.

**Comment by cladmi** [comment]
I can do the split and push it here if you want.

**Comment by cladmi** [comment]
I reviewed and tested the changes but I also contributed by splitting the commits so I cannot ACK.

Can someone else ACK the changes ?

**Comment by gebart** [comment]
Thanks @cladmi

**Comment by miri64** [comment]
Tested with both `gcc` and `clang` on Ubuntu 16.04 and Ubuntu 18.04 (riotdocker).

**Pull request merged**